src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -281,19 +283,89 @@ private void StartAnalysis(IDependencyChainNode<PythonAnalyzerEntry> node, int v
                 var analysis = new DocumentAnalysis((IDocument)module, version, walker.GlobalScope, walker.Eval);
 
                 (module as IAnalyzable)?.NotifyAnalysisComplete(analysis);
-                node.Value.TrySetAnalysis(analysis, version, _syncObj);
+                entry.TrySetAnalysis(analysis, version);
+                node.Commit();
 
-                _log?.Log(TraceEventType.Verbose, $"Analysis of {module.Name}({module.ModuleType}) complete in {(stopWatch.Elapsed - startTime).TotalMilliseconds} ms.");
+                _log?.Log(TraceEventType.Verbose, $"Analysis of {module.Name}({module.ModuleType}) is completed in {(stopWatch.Elapsed - startTime).TotalMilliseconds} ms.");
```

Comment:
Nit: "Analysis of _ completed in _" is more grammatically correct ("analysis of _ complete in _" was also fine).